File: web/src/api/signalCenter.ts
import apiClient from './index';
import { toCamelCase } from './utils';

export type SignalMarket = 'CN' | 'US';
export interface SignalSummary {
  market: SignalMarket; signalDate: string; status: 'pending' | 'completed' | 'failed' | 'skipped';
  selectedSymbol: string | null; decision: 'BUY' | 'NO_TRADE' | null; confidence: string | null;
  createdAt: string; completedAt: string | null;
}
export interface SignalDetail extends SignalSummary {
  analysis: { thesis: string; positiveSignals: string[]; risks: string[]; invalidations: string[] } | null;
  candidateSnapshot: {
    capturedAt: string; candidates: Array<{ symbol: string }>;
    sourceAvailability: Record<string, { status: string; dataAsOf: string | null; generatedAt: string | null }>;
  };
  model: string | null; promptVersion: string; error: string | null;
}
export interface DailySignals { items: SignalDetail[]; requestedDates: Record<SignalMarket, string> }
const base = '/api/v1/signal-center';
export const signalCenterApi = {
  async daily(signalDate?: string): Promise<DailySignals> {
    const { data } = await apiClient.get(`${base}/daily`, { params: { signal_date: signalDate } });
    // Market identifiers are dictionary keys, not snake_case field names.
    return { items: toCamelCase(data.items), requestedDates: data.requested_dates };
  },
  async history(offset = 0): Promise<SignalSummary[]> {
    return toCamelCase((await apiClient.get(`${base}/history`, { params: { offset, limit: 50 } })).data);
  },
  async detail(market: SignalMarket, signalDate: string): Promise<SignalDetail> {
    return toCamelCase((await apiClient.get(`${base}/${market}/${signalDate}`)).data);
  },
};

File: src/finance_analysis/interfaces/api/v1/schemas/signal_center.py
"""Explicit authenticated read contract for immutable daily research decisions."""

from datetime import date, datetime
from typing import Any, Literal
from pydantic import BaseModel
from finance_analysis.signal_center.prompt import Decision

Market = Literal["CN", "US"]


class ReturnHorizon(BaseModel):
    days: Literal[1, 3, 5, 10]
    target_date: date
    status: Literal["pending", "missing", "available"]
    value: float | None


class SignalEvaluation(BaseModel):
    method: Literal["next_session_open_v1"]
    status: Literal["not_applicable", "pending", "partial", "complete", "unavailable"]
    reason: str | None = None
    entry_date: date | None = None
    entry_price: float | None = None
    as_of: date | None = None
    observed_sessions: int = 0
    missing_dates: list[date] = []
    horizons: list[ReturnHorizon] = []
    mfe: float | None = None
    mae: float | None = None
    max_drawdown_close: float | None = None
    evaluated_at: datetime


class SignalSummary(BaseModel):
    market: Market
    signal_date: date
    status: Literal["pending", "completed", "failed", "skipped"]
    selected_symbol: str | None = None
    decision: Literal["BUY", "NO_TRADE"] | None = None
    confidence: Literal["low", "medium", "high"] | None = None
    created_at: datetime
    completed_at: datetime | None = None
    evaluation: SignalEvaluation


class SignalDetail(SignalSummary):
    analysis: Decision | None = None
    candidate_snapshot: dict[str, Any]
    model: str | None = None
    backend: str | None = None
    prompt_version: str
    screening: list[dict[str, Any]] | None = None
    error: str | None = None


class DailySignals(BaseModel):
    items: list[SignalDetail]
    requested_dates: dict[Market, date]
